contracts/oracles/SFrxETHOracle.sol

```diff
@@ -1,29 +1,101 @@
  * @notice This oracle fetches the price of sfrxETH
  */
-contract SFrxETHOracle is CorrelatedTokenOracle {
+contract SFrxETHOracle is AccessControlledV8, OracleInterface {
+    /// @notice Address of SfrxEthFraxOracle
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ISfrxEthFraxOracle public immutable SFRXETH_FRAX_ORACLE;
+
+    /// @notice Address of sfrxETH
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable SFRXETH;
+
+    /// @notice Maximum allowed price difference
+    uint256 public maxAllowedPriceDifference;
+
+    /// @notice Emits when the maximum allowed price difference is updated
+    event MaxAllowedPriceDifferenceUpdated(uint256 oldMaxAllowedPriceDifference, uint256 newMaxAllowedPriceDifference);
+
+    /// @notice Thrown if the price data is invalid
+    error BadPriceData();
+
+    /// @notice Thrown if the price difference exceeds the allowed limit
+    error PriceDifferenceExceeded();
+
+    /// @notice Thrown if the token address is invalid
+    error InvalidTokenAddress();
+
     /// @notice Constructor for the implementation contract.
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(
-        address sfrxETH,
-        address frxETH,
-        address resilientOracle
-    ) CorrelatedTokenOracle(sfrxETH, frxETH, resilientOracle) {}
+    constructor(address _sfrxEthFraxOracle, address _sfrxETH) {
+        ensureNonzeroAddress(_sfrxEthFraxOracle);
+        ensureNonzeroAddress(_sfrxETH);
+
+        SFRXETH_FRAX_ORACLE = ISfrxEthFraxOracle(_sfrxEthFraxOracle);
+        SFRXETH = _sfrxETH;
+
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Sets the contracts required to fetch prices
+     * @param _accessControlManager Address of the access control manager contract
+     * @param _maxAllowedPriceDifference Maximum allowed price difference
+     */
+    function initialize(address _accessControlManager, uint256 _maxAllowedPriceDifference) external initializer {
+        ensureNonzeroValue(_maxAllowedPriceDifference);
+
+        __AccessControlled_init(_accessControlManager);
+        maxAllowedPriceDifference = _maxAllowedPriceDifference;
+    }
+
+    /**
+     * @notice Sets the maximum allowed price difference
+     * @param _maxAllowedPriceDifference Maximum allowed price difference
+     */
+    function setMaxAllowedPriceDifference(uint256 _maxAllowedPriceDifference) external {
+        _checkAccessAllowed("setMaxAllowedPriceDifference(uint256)");
+        ensureNonzeroValue(_maxAllowedPriceDifference);
+
+        emit MaxAllowedPriceDifferenceUpdated(maxAllowedPriceDifference, _maxAllowedPriceDifference);
+        maxAllowedPriceDifference = _maxAllowedPriceDifference;
+    }
 
     /**
-     * @notice Gets the frxETH for 1 sfrxETH
-     * @return amount Amount of frxETH
+     * @notice Fetches the USD price of sfrxETH
+     * @param asset Address of the sfrxETH token
+     * @return price The price scaled by 1e18
```

Comment:
Can we also add the revert error netspec too?

Reply:
https://github.com/VenusProtocol/oracle/pull/191/commits/36f63b5e1199f7da589780357e9a7772eac72e2f